src/Desktop.Analyzers/UnitTests/DoNotUseInsecureDTDProcessingXmlDocumentConstructedWithNoSecureResolverTests.cs

```diff
@@ -0,0 +1,874 @@
+            permission.AddPermission(NetworkAccess.Connect, ""http://litwareinc.com/data/"");
+            myPermissions.SetPermission(permission);
+            XmlSecureResolver resolver = new XmlSecureResolver(new XmlUrlResolver(), myPermissions);
+
+            XmlDocument doc = new XmlDocument();
+            doc.XmlResolver = resolver;
+        }
+    }
+}"
+            );
+
+            VerifyBasic(@"
+Imports System.Net
+Imports System.Security
+Imports System.Security.Permissions
+Imports System.Xml
+
+Namespace TestNamespace
+    Class TestClass
+        Private Shared Sub TestMethod()
+            Dim myPermissions As New PermissionSet(PermissionState.None)
+            Dim permission As New WebPermission(PermissionState.None)
+            permission.AddPermission(NetworkAccess.Connect, ""http://www.contoso.com/"")
+            permission.AddPermission(NetworkAccess.Connect, ""http://litwareinc.com/data/"")
+            myPermissions.SetPermission(permission)
+            Dim resolver As New XmlSecureResolver(New XmlUrlResolver(), myPermissions)
+
+            Dim doc As New XmlDocument()
+            doc.XmlResolver = resolver
+        End Sub
+    End Class
+End Namespace"
+            );
+        }
+
+        [Fact]
+        public void XmlDocumentSetResolverToNullInTryClauseShouldNotGenerateDiagnostic()
+        {
+            VerifyCSharp(@"
+using System.Xml;
+
+namespace TestNamespace
+{
+    class TestClass
+    {
+        private static void TestMethod()
+        {
+            XmlDocument doc = new XmlDocument();
+            try
+            {
+                doc.XmlResolver = null;
+            }
+            catch { throw; }
+        }
+    }
+}"
+            );
+
+            VerifyBasic(@"
+Imports System.Xml
+
+Namespace TestNamespace
+    Class TestClass
+        Private Shared Sub TestMethod()
+            Dim doc As New XmlDocument()
+            Try
+                doc.XmlResolver = Nothing
+            Catch
+                Throw
+            End Try
+        End Sub
+    End Class
+End Namespace"
+            );
+        }
+
+        [Fact]
+        public void XmlDocumentNoResolverShouldGenerateDiagnostic()
+        {
+            VerifyCSharp(@"
+using System.Xml;
+
+namespace TestNamespace
+{
+    class TestClass
+    {
+        private static void TestMethod()
+        {
+            XmlDocument doc = new XmlDocument();
+        }
+    }
+}",
+                GetCA3075XmlDocumentWithNoSecureResolverCSharpResultAt(10, 25, "TestMethod")
+            );
+
+            VerifyBasic(@"
+Imports System.Xml
+
+Namespace TestNamespace
+    Class TestClass
+        Private Shared Sub TestMethod()
+            Dim doc As New XmlDocument()
+        End Sub
+    End Class
+End Namespace",
+                GetCA3075XmlDocumentWithNoSecureResolverBasicResultAt(7, 17, "TestMethod")
+            );
+        }
+
+        [Fact]
+        public void XmlDocumentNoResolverAsFieldShouldGenerateDiagnostic()
```

Comment:
This is the same test as on Line 158, `XmlDocumentAsFieldNoResolverShouldGenerateDiagnostic`.